iceoryx_posh/include/iceoryx_posh/runtime/posh_runtime.hpp

```diff
@@ -51,6 +51,8 @@ class RunnableData;
 class PoshRuntime
 {
   public:
+    using factory_t = std::function<PoshRuntime&(const ProcessName_t&)>;
```

Comment:
should be doable by a plain C function pointer

Reply:
@elBoberido only lambdas which capture something. But a `cxx::function_ref<PoshRuntime&(const ProcessName_t&)>` could maybe work out of the box. @FerdinandSpitzschnueffler  could you try it?

Reply:
Using function_ref will not work for storing alone. It must be combined with some storage object, which could be a function pointer here if we don not capture anything. Lambdas cannot work in a generic way, they each have a distinct type.

Reply:
Give me some time until next week and I also have a viable alternative to std::function for this (and other scenarios). So if it is not easy to change right now, no big deal. :-)

Reply:
Since a C function pointer cannot be used (we have a capturing lambda in posh_runtime_single_process) and cxx::function_ref does not work out of the box (as @MatthiasKillat described below) I suggest to merge the PR and replace the std::function by @MatthiasKillat's alternative.

Reply:
I can live with that, especially since it already was a std::function before your fix